ros_buildfarm/templates/snippet/builder_system-groovy_check-for-failing-jobs.xml.em

```diff
@@ -0,0 +1,82 @@
+@(SNIPPET(
+    'builder_system-groovy',
+    command=
+"""// CHECK FOR FAILING JOBS
+import hudson.model.Job
+
+// Control the number of failures in a row to check
+min_number_of_failures = 5
+// Control the minimun amount of time failing
+min_days_failing = 7
```

Comment:
I'd like to change the paradigm of leaving packages failing on the farm to catching bad releases and rolling them back. Leaving the bad releases on the farm blocks other builds and releases from being tested and can get us into a bad state. A rollback of the new release will restore the ability to build the old version, and just like leaving it broken the old version will be available for users. I can't think of a time when a package was failing for 3+days and then recovered without the maintainer doing a new release. So we might as well roll them back in the mean time. It saves resources and unblocks others. 

What would be really useful would be to know the profile of builds of one package across each arch. 
| Package | amd64 | arm64 | 
| foo | :+1:  |   :-1: | 
| bar | :-1: | :-1: |

And the other valuable thing would be to have grouping by packages in a repo as that's the unit of release. So I can judge the successrate of a release. Often regressions are correlated across multiple packages in a release. And having them grouped by repo would be valuable too.


Possibly 

REPONAME | build type1 | build type 2| ...| 
package_name1 | fail_cout | fail_count | ... |
package_name2 | fail_cout | fail_count | ... |
package_name3 | fail_cout | fail_count | ... |

Thus at a very quick glance it would be easy to say repo X has a problem let me find the last release to review. 

And we can filter it to something like 3 failed builds in a row for any package in the repo. Basically enough to keep out an intermittent failures if a build fails for something like an upstream invalidation etc.